mix_utils/utils.py

```diff
@@ -0,0 +1,35 @@
+from typing import Callable
+
+class CheckDuplicate():
+    """It aims to avoid duplication in the JSON/CASE file generated by the parsers (UFED, AXIOM etc.)
+    """
+    
+    def check_value(self, *args: tuple[str, ...], value: str, list_values: list[str],
+            list_objects: list[dict], observable_generating_f: Callable[..., dict]) -> dict:
```

Comment:
The `Callable` at the end of this argument list looks a little difficult to understand.  Is the last type in the `Callable` type specialization meant to be the `**kwargs` argument?

A pattern I've seen, and I forget if it's officially prescribed somewhere, is that if there are positional arguments and keyword arguments with allowance for extras, the signature to use is:

```python
def f(arg1: int, arg2: str, *args: typing.Any, kwarg1: bool, **kwargs: typing.Any) -> str: ...
```

That is, `typing.Any` should be the type of the arbitrary positional and keyword arguments.

This might be evident if you try using this function in a test script that is strictly type-reviewed (e.g.[^1] `mypy --strict`).

I actually like a test accompany this function, because I now am not understanding why `check_value` is wrapped in a class.  Is there state the class is trying to carry and share between its methods?  It looks like `check_value` can just be a module-level function.

Also, the subscript operator on Python built-ins, to act as type signatures, doesn't work in Python 3.8, but does in Python 3.9 and later.  See e.g. the differences in the first code snippet in this docs section:
* https://docs.python.org/3.8/library/typing.html#type-aliases - 3.8
* https://docs.python.org/3.9/library/typing.html#type-aliases - 3.9

If this library is to support Python 3.8 projects, you'll need to use `typing.Tuple[str, ...]` instead of `tuple[str, ...]`, at least until October of this year when Python 3.8 goes End-of-Life.  Schedule is here:
https://www.python.org/downloads/

[^1]: Disclaime: Participation by NIST in the creation of the documentation of mentioned software is not intended to imply a recommendation or endorsement by the National Institute of Standards and Technology, nor is it intended to imply that any specific software is necessarily the best available for the purpose.

Reply:
The observable_generating_f argument is a function whose parameters are specified in *args, because they change depending on the value of the observable_generating_f value. I saw an example of these kind in the mypy tool documentation (https://mypy.readthedocs.io/en/stable/kinds_of_types.html#callable-types-and-lambdas). It represents a function with a dynamic number of arguments that returns a dict.

Reply:
- "... the check_value can just be a module-level function." Yes it can be  a module at function level. The intention was to put in a module the utilities shared by all parsers, actually the only shared code is related to avoid duplication. Each check is carried out by different methods within the XXXtoJSON.py module (XXX=UFED or AXIOM etc.), so check_value can be defined as a module at function level. 
- I'm preparing the code for pytest and I'll create another PR (hopefully the last one)
-  I can replace tuple[str, ...] with typing.Tuple[str, ...] 
- as for the alternative to Callable, I'm going over the mypy documentation again to find a clearer declaration

Reply:
Oh, I didn't mean to suggest the `Callable` was wrong.  Just that if you want to use the `*args, **kwargs` style of catching extra arguments, the `Callable` signature shouldn't have a `dict` in it.  It can return a `dict`, that's fine, but the function abstraction shouldn't have a `dict` as an argument.

Please do add the `pytest` to this PR.  I've stubbed a toe a few times thinking I'd gotten the type signatures great and even elegant looking, only for the runtime to flop immediately.

Reply:
In this specific case I don't use dict among the arguments of the Callable (function). 
I have prepared the code for pytest, I need only to add another couple of functions to cover all cases, by tomorrow morning I'll raise the new PR. Thanks for your support!

Reply:
I've type-reviewed `/mix_utils` with your new updates, so I'm closing this thread as resolved.